systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -0,0 +1,162 @@
+     * - new StatefulSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) has priority over Kafka specification.
+     * 7. - Produce and consume messages in given Kafka Cluster.
+     * - Clients can produce and consume messages.
+     *
+     * @usecase
+     * - kafka-node-pool
+     */
+    @ParallelNamespaceTest
+    void testKafkaManagementTransferToAndFromKafkaNodePool(ExtensionContext extensionContext) {
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(200)
+            .withNamespaceName(clusterOperator.getDeploymentNamespace())
+            .build();
+
+        LOGGER.info("Deploying Kafka Cluster: {}/{} controlled by KafkaNodePool: {}", testStorage.getNamespaceName(), testStorage.getClusterName(), testStorage.getKafkaNodePoolName());
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 3, 1).build());
+        LOGGER.info("Kafka Cluster: {}/{} is ready", testStorage.getNamespaceName(), testStorage.getClusterName());
+
+        // increase number of kafka replicas in KafkaNodePool
+        LOGGER.info("Modifying KafkaNodePool: {}/{} by increasing number of Kafka replicas from '3' to '4'", testStorage.getNamespaceName(), testStorage.getKafkaNodePoolName());
+        KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getKafkaNodePoolName(), kafkaNodePool -> {
+            kafkaNodePool.getSpec().setReplicas(4);
+        }, testStorage.getNamespaceName());
+
+        waitForPodSetKafkaPodsReady(testStorage.getNamespaceName(), 4, testStorage.getKafkaStatefulSetName());
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResource(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+        LOGGER.info("Transferring Kafka Cluster: {}/{} from KafkaNodePool enabled, to being disabled by removing annotation", testStorage.getNamespaceName(), testStorage.getClusterName());
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> {
+                kafka.getMetadata().getAnnotations().clear();
```

Comment:
By this invocation, you delete all annotations and also the operator one AFAIK. I don't think that's what we want. You should remove a specific annotation related to KafkaNodePool instead of this whole clear.

Reply:
there is no other annotation than the one regarding enabling management by node pool. 
Kafka itself does not need annotation you mentioned unless you have two cluster operators watching namespace Kafka would exist in.